commit edd0a6e9b22c8edad94b9e8a6ca370912fbeb642
Author: popcornmix <popcornmix@gmail.com>
Date:   Wed Jan 20 17:50:09 2016 +0000

    smsx95xx: Add option to disable the crimes against truesize fix
    
    It may improve iperf numbers on Pi 1, but may generate dmesg warnings and possibly cause network issues
    See issue 1248.

diff --git a/drivers/net/usb/smsc95xx.c b/drivers/net/usb/smsc95xx.c
old mode 100755
new mode 100644
index 7483222..a61bd08
--- a/drivers/net/usb/smsc95xx.c
+++ b/drivers/net/usb/smsc95xx.c
@@ -75,6 +75,10 @@ static bool turbo_mode = false;
 module_param(turbo_mode, bool, 0644);
 MODULE_PARM_DESC(turbo_mode, "Enable multiple frames per Rx transaction");
 
+static bool truesize_mode = false;
+module_param(truesize_mode, bool, 0644);
+MODULE_PARM_DESC(truesize_mode, "Report larger truesize value");
+
 static char *macaddr = ":";
 module_param(macaddr, charp, 0);
 MODULE_PARM_DESC(macaddr, "MAC address");
@@ -1841,6 +1845,8 @@ static int smsc95xx_rx_fixup(struct usbnet *dev, struct sk_buff *skb)
 				if (dev->net->features & NETIF_F_RXCSUM)
 					smsc95xx_rx_csum_offload(skb);
 				skb_trim(skb, skb->len - 4); /* remove fcs */
+				if (truesize_mode)
+					skb->truesize = size + sizeof(struct sk_buff);
 
 				return 1;
 			}
@@ -1858,6 +1864,8 @@ static int smsc95xx_rx_fixup(struct usbnet *dev, struct sk_buff *skb)
 			if (dev->net->features & NETIF_F_RXCSUM)
 				smsc95xx_rx_csum_offload(ax_skb);
 			skb_trim(ax_skb, ax_skb->len - 4); /* remove fcs */
+			if (truesize_mode)
+				ax_skb->truesize = size + sizeof(struct sk_buff);
 
 			usbnet_skb_return(dev, ax_skb);
 		}
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1158_edd0a6e9b22c8edad94b9e8a6ca370912fbeb642.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1158_edd0a6e9b22c8edad94b9e8a6ca370912fbeb642.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
